src/cli2/common.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+
+import {join as pathJoin} from "path";
+import fs from "fs-extra";
+
+import type {PluginDirectoryContext} from "./cliPlugin";
+import {parse as parseConfig, type InstanceConfig} from "./instanceConfig";
+
+export async function loadInstanceConfig(
+  baseDir: string
+): Promise<InstanceConfig> {
+  const projectFilePath = pathJoin(baseDir, "sourcecred.json");
+  const contents = await fs.readFile(projectFilePath);
+  return Promise.resolve(parseConfig(JSON.parse(contents)));
```

Comment:
Yes, but only because of a JavaScript [design flaw][1] that has really
aged like milk with the prevalence of TypeScript and Flow. I prefer to
write the code in the sensible, principled style, even though it’s not
strictly required by the language.

[1]: https://github.com/promises-aplus/promises-spec/issues/94#issuecomment-16176966

Reply:
Certainly won't argue with you that Promises have a bunch of design flaws. I've built several client projects with https://github.com/fluture-js/Fluture instead of Promises and there's a lot of great properties there that I'm sorely missing having gone back to Promises.

On the other hand, the language integration definitely yields productivity benefits over going against the unfortunate standard, and reduces learning curve.

In this case `return Promise.resolve()` is a pattern that makes use of one of those Promise design choices, to silently unpack Promises at every opportunity.

In other words:
```js
Promise.resolve(Promise.resolve(123))
> Promise { <state>: "fulfilled", <value>: 123 }
```
We get a `Promise<number>` not a `Promise<Promise<number>>`.
A surprising and undesirable property if you ask me.

My taste preference here is to avoid that surprise, so I generally encourage others not to `return Promise.resolve()`.

Anyhow, if you feel like there's a benefit to going the other way, curious to hear about it. But for this PR, I'm fine with either :]

Reply:
Personally, I'm not a fan of this style. The linked comment does not explain why this is a design flaw, and I'm used to the semantics of just returning things from async functions and getting promises. I'm not going to ask for this to change, since it seems correct either way (see NHIWHDI), but in code that I write I'll continue to use the same idiom I've used in the past.

Reply:
I prefer the non-redundant style, although haven't invested the effort to really understand why William prefers it. Since we don't have a lint rule to automatically enforce it, I say it's authors pick on which to use.

Reply:
That `resolve(resolve(123))` is equivalent to `resolve(123)` is
precisely the design flaw that I’m trying to avoid. If the standard
didn’t have that flaw, then you would have to return `resolve(123)`
here, so returning `123` is surfacing the flaw, not hiding it. When
I read `async function foo() { return 123; }`, which shouldn’t compile,
I have to remind myself that it works because of this flaw.

I’m happy to discuss in more detail, and I can provide a specific
example of code that this design flaw makes it impossible to write
cleanly (forcing ugly contortions in the form of boxing) as well as
point to some of the complexity that the flaw forces in real code. I’ll
keep it as is for now, and am fine with it being author’s pick, just as
so many other things are.